src/components/PrioritySelector/__tests__/PrioritySelector.spec.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { render, RenderResult, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { userEvent } from '@testing-library/user-event';
+import { Priority, PriorityItemType, PrioritySelector } from '../PrioritySelector';
+
+const priorityList: PriorityItemType<string>[] = [
+    { priority: Priority.Mandatory, label: 'Mandatory', value: 'required' },
+    { priority: Priority.Important, label: 'Important', value: 'strongly_favored' },
+    { priority: Priority.Optional, label: 'Optional', value: 'favored' },
+    { priority: Priority.Exclude, label: 'Exclude', value: 'banned' },
+];
+
+const priorityListRandom: PriorityItemType<string>[] = [
+    { priority: Priority.Optional, label: 'Optional', value: 'favored' },
+    { priority: Priority.Mandatory, label: 'Mandatory', value: 'required' },
+    { priority: Priority.Exclude, label: 'Exclude', value: 'banned' },
+    { priority: Priority.Important, label: 'Important', value: 'strongly_favored' },
+];
+
+describe('PrioritySelector', () => {
+    let view: RenderResult;
+    const onSelectMock = jest.fn();
+
+    beforeEach(() => {
+        view = render(
+            <PrioritySelector
+                selectedItem={priorityList[0]}
+                list={priorityList}
+                onSelect={onSelectMock}
+            />
+        );
+    });
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('renders correctly with all props provided', () => {
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('button', { name: 'Mandatory' })).toBeInTheDocument();
+    });
+
+    it('should render correctly opened', async () => {
+        const user = userEvent.setup();
+        await user.click(screen.getByRole('button', { name: 'Mandatory' }));
+        expect(screen.getAllByRole('option')).toHaveLength(4);
```

Comment:
maybe add a snapshot here too, just for completeness, to have the popup as well as the trigger.